src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx

```diff
@@ -78,11 +81,11 @@ function NotificationEmailCard(props: Props) {
   const buttonText = useMemo(() => {
     if (!emailProp && !unconfirmedEmail && !hasConfirmEmail) {
       return t('settings.notifications.email.button_submit_label')
-    } else if (unconfirmedEmail) {
+    } else if (unconfirmedEmail && unconfirmedEmail === email) {
```

Comment:
Would you mind adding tests for the new conditions?

Reply:
We have that test. 

https://github.com/decentraland/account/blob/ebe4e22827b91ca397d708696b5701a4564c5cef/src/components/Notifications/NotificationEmailCard/NotificationEmailCard.spec.tsx#L125